src/directives/formly-form.test.js

```diff
@@ -933,12 +933,19 @@ describe('formly-form', () => {
 
       const listener = sinon.spy()
       const expression = sinon.spy()
+      const deepLinkField = getNewField()
+      deepLinkField.key = 'foo.bar'
+      deepLinkField.watcher = {
+        listener,
+        expression: sinon.spy(),
```

Comment:
Sorry, I think I told you wrong. The `expression` shouldn't be here. We want to test behavior for when an expression is omitted. So this should be:

```javascript
deepLinkField.watcher = {
  listener: sinon.spy(),
  // no expression
}
```

Then we need to manually change the value of `foo.bar`, run `scope.$digest()` and validate that the `listener` was called with the new value of `foo.bar`. I suggest that this all be put into its own `it` statement.